tests/Unit/Domain/CoordinateMaps/Test_UniformCylindricalSide.cpp

```diff
@@ -0,0 +1,518 @@
+                     (1.0 - z_plane_frac_plus_one));
+
+    // Compute the minimum allowed value of the angle alpha_minus.
+    // (these quantities are measured from zero; note the minus signs)
+    const double theta_max_minus_one = acos(-z_plane_frac_minus_one);
+    const double theta_max_minus_two =
+        acos(-(z_plane_minus_two - center_two[2]) / radius_two);
+    const double min_alpha_minus = 1.1 * theta_max_minus_one;
+    // max_radius_one_from_alpha comes from the restriction
+    // that alpha > min_alpha_minus
+    // and the expression for z_plane_frac_plus_one (used to eliminate
+    // center_one[2]) and the expression for z_plane_frac_minus_one.
+    const double max_radius_one_from_alpha =
+        (z_plane_plus_one - z_plane_minus_two +
+         radius_two * tan(min_alpha_minus) * sin(theta_max_minus_two)) /
+        (tan(min_alpha_minus) * sin(theta_max_minus_one) +
+         z_plane_frac_plus_one - z_plane_frac_minus_one);
+    const double max_radius_one = std::min(
+        {max_radius_one_planes, max_radius_one_fit, max_radius_one_from_alpha});
+    const double min_radius_one = 0.08 * radius_two;
+    CHECK(max_radius_one >= min_radius_one);
+    return min_radius_one + unit_dis(gen) * (max_radius_one - min_radius_one);
+  }();
+  CAPTURE(radius_one);
+
+  const std::array<double, 3> center_one = {
+      center_two[0], center_two[1],
+      z_plane_plus_two - z_plane_frac_plus_one * radius_one};
+  CAPTURE(center_one);
+
+  const double z_plane_minus_one =
+      center_one[2] + radius_one * z_plane_frac_minus_one;
+  CAPTURE(z_plane_minus_one);
+
+  if (flip_z_axis) {
+    // Here we test the map with z_plane_minus_one equal to
+    // z_plane_minus_two.  We do this by simply flipping the map
+    // parameters about the z axis (and exchanging parameters named
+    // plus and minus).  This way we don't need to rewrite the entire
+    // test for z_plane_minus_one==z_plane_minus_two.
+    const CoordinateMaps::UniformCylindricalSide map(
+        {{center_one[0], center_one[1], -center_one[2]}},
+        {{center_two[0], center_two[1], -center_two[2]}},
+        radius_one,
+        radius_two, -z_plane_minus_one, -z_plane_plus_one, -z_plane_minus_two,
+        -z_plane_plus_two);
+    test_suite_for_map_on_cylinder(map, 1.0, 2.0, true, true);
+  } else {
+    const CoordinateMaps::UniformCylindricalSide map(
+        center_one, center_two, radius_one, radius_two, z_plane_plus_one,
+        z_plane_minus_one, z_plane_plus_two, z_plane_minus_two);
+    test_suite_for_map_on_cylinder(map, 1.0, 2.0, true, true);
+  }
+}
+
+void test_uniform_cylindrical_side() {
+  INFO("UniformCylindricalSide");
+
+  // Set up random number generator
+  MAKE_GENERATOR(gen);
+  std::uniform_real_distribution<> unit_dis(0.0, 1.0);
+  std::uniform_real_distribution<> interval_dis(-1.0, 1.0);
+  std::uniform_real_distribution<> angle_dis(0.0, 2.0 * M_PI);
+
+  // Choose some random center for sphere_two
```

Comment:
Seems like there's some code duplication between here and the previous function. Any reason not to factor to anonymous namespace the parts of these functions (setting up the random values etc.) that are the same for both?

Reply:
I could refactor these a bit, but there is very little code that is the same.  Just setting up center and radius of sphere two.

And I fear that these functions might diverge even more once the maps are combined into a domain and I need to change the allowed ranges of parameters again (your question farther above).